CoreTableView/src/au/gov/asd/tac/constellation/views/tableview2/TableViewPane.java

```diff
@@ -1239,44 +1234,99 @@ public void updateSelection(final Graph graph, final TableViewState state) {
                     final ReadableGraph readableGraph = graph.getReadableGraph();
-                    try {
-                        final boolean isVertex = state.getElementType() == GraphElementType.VERTEX;
-                        final int selectedAttributeId = isVertex
-                                ? VisualConcept.VertexAttribute.SELECTED.get(readableGraph)
-                                : VisualConcept.TransactionAttribute.SELECTED.get(readableGraph);
-                        final int elementCount = isVertex
-                                ? readableGraph.getVertexCount()
-                                : readableGraph.getTransactionCount();
-                        for (int elementPosition = 0; elementPosition < elementCount; elementPosition++) {
-                            final int elementId = isVertex
-                                    ? readableGraph.getVertex(elementPosition)
-                                    : readableGraph.getTransaction(elementPosition);
-                            boolean isSelected = false;
-                            if (selectedAttributeId != Graph.NOT_FOUND) {
-                                isSelected = readableGraph.getBooleanValue(selectedAttributeId, elementId);
-                            }
-                            if (isSelected) {
-                                selectedIds.add(elementId);
-                            }
-                        }
-                    } finally {
-                        readableGraph.release();
-                    }
+                    addToSelectedIds(selectedIds, readableGraph, state);
 
                     // update table selection
                     final int[] selectedIndices = selectedIds.stream().map(id -> elementIdToRowIndex.get(id))
                             .map(row -> table.getItems().indexOf(row)).mapToInt(i -> i).toArray();
-
-                    Platform.runLater(() -> {
+                    
+                    Platform.runLater(() -> {                        
                         selectedProperty.removeListener(tableSelectionListener);
                         table.getSelectionModel().clearSelection();
                         if (!selectedIds.isEmpty()) {
                             table.getSelectionModel().selectIndices(selectedIndices[0], selectedIndices);
                         }
-                        selectedProperty.addListener(tableSelectionListener);
+                        selectedProperty.addListener(tableSelectionListener);                            
                     });
                 }
             }
         }
     }
+       
+    /**
+     * A version of the updateSelection(Graph, TableViewState) function which is
+     * to be run on the JavaFX Application Thread
+     * 
+     * @param graph the graph to read selection from.
+     * @param state the current table view state.
+     */
+    private void updateSelectionFromJAT(final Graph graph, final TableViewState state) {
```

Comment:
A real minor one but how about calling it `updateSelectionFromFxThread`?

Reply:
done